scripts/00_training/sparse_parities.py

```diff
@@ -0,0 +1,230 @@
+# %%
+from accelerate import Accelerator
+from accelerate.utils import set_seed
+
+import torch
+import torch.nn as nn
+
+import evaluate
+import numpy as np
+import wandb
+import os
+import json
+import argparse
+
+from src.data import SparseParity
+from src.model import MLP, custom_weight_init
+from src.metrics import get_metrics_mlp
+
+# %%
+# replicating https://arxiv.org/pdf/2207.08799.pdf
+
+DATASET_NAME = "sparse-parities"
+
+
+# %%
+class HingeLoss(torch.nn.Module):
+    def __init__(self):
+        super(HingeLoss, self).__init__()
+
+    def forward(self, output, target):
+        hinge_loss = 1 - torch.mul(torch.squeeze(output), torch.squeeze(target))
+        hinge_loss[hinge_loss < 0] = 0
+        return hinge_loss
+
+
+def get_dataloaders(train_size, test_size, train_bsz, test_bsz):
+    data = SparseParity(train_size + test_size, 40, 3)
+    train, test = torch.utils.data.random_split(data, [train_size, test_size])
```

Comment:
should avoid hard-coding the constants for SparseParity - either include them as arguments to the function or set them as default argument values in the constructor for SparseParsity